askama_shared/src/parser.rs

```diff
@@ -404,17 +404,34 @@ fn variant_path(i: &[u8]) -> IResult<&[u8], MatchVariant> {
     })(i)
 }
 
-fn target_single(i: &[u8]) -> IResult<&[u8], Target> {
-    map(identifier, |s| Target::Name(s))(i)
-}
+fn target(i: &[u8]) -> IResult<&[u8], Target> {
+    let mut opt_opening_paren = map(opt(ws(tag("("))), |o| o.is_some());
+    let mut opt_closing_paren = map(opt(ws(tag(")"))), |o| o.is_some());
+
+    let (i, target_is_tuple) = opt_opening_paren(i)?;
+    if target_is_tuple {
+        let (i, is_empty_tuple) = opt_closing_paren(i)?;
+        if is_empty_tuple {
+            return Ok((i, Target::Tuple(vec![])));
+        }
 
-fn target_tuple(i: &[u8]) -> IResult<&[u8], Target> {
-    let parts = separated_list0(tag(","), ws(identifier));
-    let trailing = opt(ws(tag(",")));
-    let mut full = delimited(tag("("), tuple((parts, trailing)), tag(")"));
+        let (i, first_target) = target(i)?;
+        let (i, is_unused_paren) = opt_closing_paren(i)?;
+        if is_unused_paren {
+            return Ok((i, first_target));
+        }
+
+        let mut targets = vec![first_target];
+        let push_target = |_, target| targets.push(target);
+        let (i, _) = tuple((
+            fold_many0(preceded(ws(tag(",")), target), (), push_target),
```

Comment:
```suggestion
        let (i, _) = tuple((
            fold_many0(preceded(ws(tag(",")), target), (), |_, target| targets.push(target)),
```

Any particular reason this isn't inlined? :)